WinImageBuilder.psm1

```diff
@@ -1041,5 +1041,110 @@ function New-WindowsCloudImage {
     Write-Host ("Image generation finished at: {0}" -f @(Get-Date))
 }
 
+function New-WindowsFromGoldenImage {
+    [CmdletBinding()]
+    Param(
+        [parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        [string]$ConfigFilePath
+    )
+
+    $windowsImageConfig = Get-WindowsImageConfig -ConfigFilePath $ConfigFilePath
+    Write-Host ("Windows online image generation started at: {0}" -f @(Get-Date))
+    Is-Administrator
+    if (!$windowsImageConfig.run_sysprep -and !$windowsImageConfig.force) {
+        throw "You chose not to run sysprep.
+            This will build an unusable Windows image.
+            If you really want to continue use the `force = true` config option."
+    }
+
+    Check-Prerequisites
+    if ($windowsImageConfig.external_switch) {
+        $switch = Get-VMSwitch -Name $windowsImageConfig.external_switch -ErrorAction SilentlyContinue
+        if (!$switch) {
+            throw "Selected vmswitch {0} does not exist" -f $windowsImageConfig.external_switch
+        }
+        if ($switch.SwitchType -ne "External" -and !$windowsImageConfig.force) {
+            throw "Selected switch {0}} is not an external
+                switch. If you really want to continue use the `force = true` flag." `
+                -f $windowsImageConfig.external_switch
+        }
+    }
+    if ($windowsImageConfig.cpu_count -gt `
+        (Get-WmiObject Win32_Processor).NumberOfLogicalProcessors) {
+        throw "CpuCores larger then available (logical) CPU cores."
+    }
+
+    try {
+        $goldenPath = Get-PathWithoutExtension $windowsImageConfig.gold_disk_path
+        $virtualDiskPath = $goldenPath + ".vhdx"
+        Execute-Retry {
+            Resize-VHD -Path $virtualDiskPath -SizeBytes $windowsImageConfig.disk_size
+        }
+
+        Mount-VHD -Path $virtualDiskPath -Passthru | Out-Null
+        Get-PSDrive | Out-Null
+
+        $driveLetterGold = ((Get-DiskImage -ImagePath $virtualDiskPath | Get-Disk | Get-Partition | Get-Volume).DriveLetter + ":")
+
+        $driveNumber = (Get-DiskImage -ImagePath $virtualDiskPath | Get-Disk).Number
+        $maxPartitionSize = (Get-PartitionSupportedSize -DiskNumber $driveNumber -PartitionNumber 1).SizeMax
+        Resize-Partition -DiskNumber $driveNumber -PartitionNumber 1 -Size $maxPartitionSize
+
+        if ($windowsImageConfig.drivers_path) {
+            Dism /Image:$driveLetterGold /Add-Driver /Driver:$windowsImageConfig.drivers_path /ForceUnsigned /Recurse
+        }
+
+        $resourcesDir = Join-Path -Path $driveLetterGold -ChildPath "UnattendResources"
+        Copy-UnattendResources -resourcesDir $resourcesDir -imageInstallationType "Server Standard"
+        Download-CloudbaseInit $resourcesDir ([string]"AMD64")

```

Comment:
Here you need to use ilenghel's work to check if the image is already an amd64 or not